### PR TITLE
Support custom config data for bookkeeper

### PR DESCRIPTION
**Pull request by tuteng**
Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

* Some users complained that some custom configurations could not be loaded into the bookkeeper configuration  so this pr fixes it

### Modifications

* Support load custom configData for bookkeeper

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml [hunk 1]
```diff
@@ -158,34 +158,31 @@ spec:
         {{- include "pulsar.bookkeeper.ledgers.storage.class" . | nindent 8 }}
     reclaimPolicy: Delete
   {{- end }}
-  {{- if .Values.bookkeeper.custom }}
```
Comment (jdmaguire) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml:161]: With this gone, I see .Values.bookkeeper.custom no longer used.

If the intent is to put all custom config in configData, we could remove [.custom from the values file](https://github.com/streamnative/charts/blob/master/charts/sn-platform/values.yaml#L879).

**Comment by tuteng** [comment]
@jdmaguire PTAL

**Comment by Huanli-Meng** [comment]
@tuteng do we need to add/update doc for this PR?

**Comment by Huanli-Meng** [comment]
> @tuteng do we need to add/update doc for this PR?

a `no-need-doc` check box is selected. Therefore, label this doc with `no-need-doc`

**Review by fantapsody — COMMENTED** [review]
I'm not sure if any user is using `.Values.bookkeeper.custom`, removing it directly looks like a breaking change, or we can still put its value under `.spec.config.custom`? @ericsyh what's your opinion?

**Comment by ericsyh** [comment]
> I'm not sure if any user is using `.Values.bookkeeper.custom`, removing it directly looks like a breaking change, or we can still put its value under `.spec.config.custom`? @ericsyh what's your opinion?

Yes, i think it will be better to keep compatibility in minor release and break on next major release.  Otherwise, using `.Values.bookkeeper.custom` on current SNP will get problem upgrade to 1.4

**Comment by tuteng** [comment]
> > I'm not sure if any user is using `.Values.bookkeeper.custom`, removing it directly looks like a breaking change, or we can still put its value under `.spec.config.custom`? @ericsyh what's your opinion?
> 
> Yes, i think it will be better to keep compatibility in minor release and break on next major release. Otherwise, using `.Values.bookkeeper.custom` on current SNP will get problem upgrade to 1.4

Fixed @fantapsody @ericsyh PTAL

**Review by jdmaguire — APPROVED** [review]
Looks good and all comments addressed.

**Pull request merged**